src/file.ts

```diff
@@ -1752,12 +1754,20 @@ class File extends ServiceObject<File> {
       md5,
     });
 
+    const progressStream = new ProgressStream();
+    if (options.onUploadProgress) {
+      progressStream.on('progress', bytesRead => {
+        options.onUploadProgress!({bytesRead});
```

Comment:
Could we also just pass back `bytesRead` as opposed to boxing it in an object? Or if there's anything else we should pass back, let's do it now.

Reply:
Done